src/common/uilocale.cpp

```diff
@@ -514,14 +514,14 @@ const wxUILocale& wxUILocale::GetCurrent()
 
 wxUILocale::wxUILocale(const wxLocaleIdent& localeId)
 {
-    if ( localeId.IsEmpty() )
+    if ( !localeId.IsEmpty() )
     {
-        wxFAIL_MSG( "Locale identifier must be initialized" );
-        m_impl = NULL;
-        return;
+        m_impl = wxUILocaleImpl::CreateForLocale(localeId);
+    }
+    else
+    {
+        m_impl = wxUILocaleImpl::CreateUserDefault();
```

Comment:
I'm not sure if it's really a good idea to use the default locale for invalid `localeId`, why do we need to do this?

And if we do change it, the documentation for this ctor must be updated because it explicitly documents the current behaviour.

Reply:
> I'm not sure if it's really a good idea to use the default locale for invalid `localeId`, why do we need to do this?

We need it to restore the previous behaviour of `wxLocale` - selecting the system language based on the default locale (and not the preferred UI language).

> And if we do change it, the documentation for this ctor must be updated because it explicitly documents the current behaviour.

As said in my general comment, I simply forgot to update the documentation. However, we can revert the change and introduce a separate static method to instantiate a `wxUILocale` instance for the default locale.